### PR TITLE
ENG-596: sessions start: allow to pass values to start

### PR DESCRIPTION
**Pull request by itayd**
also allow watch/log to emit just the prints

### CODE REVIEW

**Review by efiShtain — APPROVED** [review]

**Review by ig-ra — APPROVED** [review]

File: cmd/ak/cmd/sessions/log.go [hunk 1]
```diff
@@ -11,10 +11,13 @@ import (
...
 
-var skip int
+var (
+	skip       int
+	justPrints bool
+)
 
 var logCmd = common.StandardCommand(&cobra.Command{
-	Use:   "log [sessions ID] [--fail] [--skip <N>] [--no-timestamps]",
+	Use:   "log [sessions ID] [--fail] [--skip <N>] [--no-timestamps] [--prints]",
```
Comment (daabr) [inline cmd/ak/cmd/sessions/log.go]: This is not the correct flag name

File: cmd/ak/cmd/sessions/log.go [hunk 2]
```diff
@@ -50,6 +53,7 @@ func init() {
 	// Command-specific flags.
 	logCmd.Flags().IntVarP(&skip, "skip", "s", 0, "number of entries to skip")
 	logCmd.Flags().BoolVarP(&noTimestamps, "no-timestamps", "n", false, "omit timestamps from track output")
```
Comment (daabr) [inline cmd/ak/cmd/sessions/log.go]:
```suggestion
	logCmd.Flags().BoolVarP(&noTimestamps, "no-timestamps", "n", false, "omit timestamps from watch output")
```

File: cmd/ak/cmd/sessions/log.go [hunk 3]
```diff
@@ -50,6 +53,7 @@ func init() {
...
+	logCmd.Flags().BoolVarP(&justPrints, "just-prints", "p", false, "print only log entries with print messages")
```
Comment (daabr) [inline cmd/ak/cmd/sessions/log.go]: "prints-only" sounds better than "just-prints", and a more accurate description

```suggestion
	logCmd.Flags().BoolVarP(&justPrints, "prints-only", "p", false, "output only workflow print messages")
```

File: cmd/ak/cmd/sessions/start.go [hunk 4]
```diff
@@ -66,6 +74,35 @@ func init() {
 	startCmd.Flags().DurationVarP(&pollInterval, "poll-interval", "i", defaultPollInterval, "watch poll interval")
 	startCmd.Flags().BoolVarP(&noTimestamps, "no-timestamps", "n", false, "omit timestamps from watch output")
 	startCmd.Flags().BoolVarP(&quiet, "quiet", "q", false, "don't print anything, just wait to finish")
+
+	startCmd.Flags().StringArrayVarP(&values, "value", "v", nil, `zero or more "key=value" pairs, where value is a JSON value`)
```
Comment (daabr) [inline cmd/ak/cmd/sessions/start.go]: 1. This technically works, but JSON may be too unfriendly to specify in the command line. Maybe specify a JSON file instead?
2. "value" is not a clear flag name. Value of what? for what? The flag description also doesn't clarify to the user the purpose and function of this flag
3. Add this to the usage string in line 24 above:

```
[--value <JSON> [...]]
```

File: cmd/ak/cmd/sessions/watch.go [hunk 5]
```diff
@@ -15,7 +15,7 @@ import (
 var endState string
 
 var watchCmd = common.StandardCommand(&cobra.Command{
-	Use:   "watch [sessions ID] [--fail] [--end-state <state>] [--timeout <duration>] [--poll-interval <duration>] [--no-timestamps] [--quiet]",
+	Use:   "watch [sessions ID] [--fail] [--end-state <state>] [--timeout <duration>] [--poll-interval <duration>] [--no-timestamps] [--quiet] [--just-prints]",
```
Comment (daabr) [inline cmd/ak/cmd/sessions/watch.go]: See my comments in `log.go`

File: cmd/ak/cmd/sessions/log.go (hunk 1 above)
Comment (itayd) [reply cmd/ak/cmd/sessions/log.go]: fixed

File: cmd/ak/cmd/sessions/log.go (hunk 2 above)
Comment (itayd) [reply cmd/ak/cmd/sessions/log.go]: fixed

File: cmd/ak/cmd/sessions/log.go (hunk 3 above)
Comment (itayd) [reply cmd/ak/cmd/sessions/log.go]: fixed

File: cmd/ak/cmd/sessions/watch.go (hunk 5 above)
Comment (itayd) [reply cmd/ak/cmd/sessions/watch.go]: fixed

File: cmd/ak/cmd/sessions/start.go (hunk 4 above)
Comment (itayd) [reply cmd/ak/cmd/sessions/start.go]: 1. agree, but in a different pr
2. will change to `input`
3. fixed

**Pull request merged**